var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -60,6 +61,19 @@ def install(self, spec, prefix):
         if spec.satisfies("@4.7.1:") and sys.platform != 'darwin':
             enabled_languages.add('go')
 
+        # Fix a standard header file for OS X Yosemite & earlier that
+        # is GCC incompatible by replacing non-GCC compliant macros
+        if sys.platform == 'darwin':
```

Comment:
> you write Yosemite and earlier, why don't you check for that (looking at versions)?

@davydden The API for OS versions isn't documented at all, AFAICT. I had some downtime so I spent a couple hours looking at the Spack code and couldn't make much heads or tails of it, so I only added a check for Yosemite. As I'm sure you're aware, the bare string comparison of OS versions will fail (e.g., `spec.architecture.platform_os.version < 10.10` is `False` when the OS version is `10.9`), so as you note in #1814, the OS version should really be a `Version` object so that it can use the comparison operators inherited from that class.

>  I am not sure this patch will work on elcapitan or sierra.

Also, AFAICT from Google searches, the string replacement doesn't affect other versions of Mac OS prior to Sierra, excluding Yosemite, of course. That said, I haven't tested this code against any versions of OS X other than Yosemite. If someone has access to OS X systems running Sierra/El Capitan/Mavericks/etc. and would be willing to test this patch, I'd appreciate it.

> So i would adjust the if so that it is in accordance with the comment.

I have made a change that I think complies with your request.